phac-nml/irida
# Delete sequence files from file storage when removed from sample

## Description of changes
Added functionality to delete files from file storage when sequence files are removed from a sample.
Note, only files that have no analysis or sequence runs associated with it will be removed from file storage.
For now, the file is remove if it's apart of a concatenated file, as cleaning up nested concatenated files gets complicated.

__Testing Instructions__

1) Test removing sequence files
- login to IRIDA
- create a sample under a project
- add sequence file(s) to sample
- verify files are on file system
- remove sequence file(s) from sample
- verify files are not file system

2) Test removing sequence files during project synchronization
- login to IRIDA
- create a sample under a project
- add sequence file(s) to sample
- verify files from source project are on file system
- initialize a project synchronization
- verify files from target project are on file system
- remove sequence file(s) from sample within source project
- verify files from source project are on file system
- synchronize the target project
- verify files from target project are on file system

Reminder to test (nested) concatenated sequence files too.

## Related issue
https://github.com/phac-nml/irida/issues/1125

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:
